plugins/ltac/tactic_debug.ml

```diff
@@ -163,13 +163,13 @@ let cvt_loc loc =
          let file = String.sub dp (lastdot+1) (dplen - (lastdot + 1)) in
          let module_name = String.sub dp 0 lastdot in
          let routine =
-           try
-             (* try text as a kername *)
-             assert (CString.is_prefix dp text);
-             let knlen = String.length text in
-             let lastdot = String.rindex text '.' in
-             String.sub text (lastdot+1) (knlen - (lastdot + 1))
-           with exn when CErrors.noncritical exn -> text
+           (* try text as a kername *)
+           if not (CString.is_prefix dp text) then text else
+             try
+               let knlen = String.length text in
+               let lastdot = String.rindex text '.' in
+               String.sub text (lastdot+1) (knlen - (lastdot + 1))
+             with exn when CErrors.noncritical exn -> text
```

Comment:
Exceptions for flow control are not great, and noncritical catch-all clauses are even worse. Use a proper bound test instead.

Reply:
(And `rindex_opt`).

Reply:
> Exceptions for flow control are not great, and noncritical catch-all clauses are even worse.

Dubious.  The bounds test is not the issue.  rindex may raise `Not_found`, which is a normal condition.  Catching all exceptions (other than assert) is often defensive coding.